guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+$this->eventDispatcher->dispatch($someEvent, $someEvent->getName());
+```
+
+This is an event that's being fired manually, which you can react upon.
+Make sure to always have a look at the event class itself in order to find out which information it contains.
+
+When subscribing to those events, your event listener method will have access to the previously created event instance. 
+
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        SomeEvent::class => 'registeringToSomeEvent'
+    ];
+}
+
+public function registeringToSomeEvent(SomeEvent $event): void
+{
+}
+```
+
+The [next section](#Specifically searching for events) will cover how to find those events without randomly stumbling upon them.
+
+### Specifically searching for events
+
+If you're really looking for a fitting event for your purpose, you might want to directly search for them.
+This can be done by searching through the `<shopware root>/platform/src` or the `<shopware root>/vendor/shopware/platform/src` directory,
+depending on whether you're using the [development](https://github.com/shopware/development) or the [production template](https://github.com/shopware/production).
+Use one of the following search terms:
+
+- `extends NestedEvent`: This way you will find the events themselves.
+- `extends Event`: This way you will find the events themselves.
+- `->dispatch`: Here you will find all the occurrences where the events are actually being fired.
+
+### Looking at the service definition
+
+Every service, that wants to fire an event sooner or later, needs access to the `event_dispatcher` in order to do so.
+
+Hence, you can have a look at all the service definitions for the [dependency injection container](../../plugin-fundamentals/dependency-injection.md)
+and therefore quickly figure out, which services and classes are having access to the said `event_dispatcher`:
+
+```xml
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="Some\Service">
+            <argument type="service" id="Another/Service"/>
+            <argument type="service" id="event_dispatcher"/>
+        </service>
+    </services>
+</container>
+```
+
+Therefore, you could simply search for occurrences of the `event_dispatcher` in the respective `.xml` files.
+
+You can also do this the other way around, by having a look at the service's constructor parameters.
+
+```php
+public function __construct(
+    Some\Service $someService,
+    EventDispatcherInterface $eventDispatcher
+) {
+    $this->someService = $someService;
+    $this->eventDispatcher = $eventDispatcher;
+}
+```
+
+If it's having access to the `EventDispatcherInterface`, you're most likely going to find at least one event being fired
```

Comment:
I know this shall not be a list but you described the pattern existing in DAL events. You have lots more pattern that you and me rely on and should be mentioned. When their naming schema at least get a line or two here it will be easier for the devs on code safari to understand them. E.g. there are often CriteriaEvents which are dispatched right before a special situation so you can alter a query. PageLoadedEvents to add data to a page in question. BusinessEvents that are used in the email sending logic and are THE event type that is used for special ecommerce events. You are THE ecommerce framework. They should get a whole paragraph. ValidationEvents that are used between HTTP requests and DAL actions to add new required form fields or remove their checks.

Reply:
Good explanation